Marketplace/music-visualizer/musicVisualizer.js

```diff
@@ -0,0 +1,268 @@
+                Messages.sendLocalMessage('Hifi-Hand-Grab', JSON.stringify({
+                    hand: "left",
+                    entityID: effectLeft
+                }));
+                Messages.sendLocalMessage('Hifi-Hand-Grab', JSON.stringify({
+                    hand: "right",
+                    entityID: effectRight
+                }));
+
+                var effectLeftProps = Entities.getEntityProperties(effectLeft, "script");
+                var effectRightProps = Entities.getEntityProperties(effectRight, "script");
+                effectLeftProps.script = TRAIL_LEFT_SCRIPT;
+                effectRightProps.script = TRAIL_RIGHT_SCRIPT;
+                Entities.editEntity(effectLeft, effectLeftProps);
+                Entities.editEntity(effectRight, effectRightProps);
+
+                var equipLeftProps = Entities.getEntityProperties(equipLeft, "visible");
+                var equipRightProps = Entities.getEntityProperties(equipRight, "visible");
+                equipLeftProps.visible = true;
+                equipRightProps.visible = true;
+                Overlays.editOverlay(equipLeft, equipLeftProps);
+                Overlays.editOverlay(equipRight, equipRightProps);
+
+            }, 700);
+
+            // once particles are equipped tablet must be auto closed
+            button.editProperties({ isActive: false });
+            tablet.webEventReceived.disconnect(onWebEventReceived);
+            tablet.gotoHomeScreen();
+            Messages.sendLocalMessage("home", HMD.homeButtonID);
+            shown = false;
+        }
+
+        if (behavior === "nobehavior") {
+            Entities.addEntity(effectJSON);
+        }
+
+        // clears audio file selection
+        updateSelection("");
+    }
+
+    // to modify HTML page's listed filename
+    function updateSelection(filename) {
+        if (filename === "") {
+            audioFile = "";
+        }
+        tablet.emitScriptEvent(JSON.stringify({
+            "file": filename
+        }));
+    }
+
+    function onWebEventReceived(event) {
+        var htmlEvent = JSON.parse(event);
+        print("Event: " + JSON.stringify(htmlEvent));
+        // Handles particle button clicks to retrieve the effect JSON from musVisLib
+        if (htmlEvent.type === "click") {
+            var effectName = htmlEvent.data;
+            var micSync = htmlEvent.sync;
+            var file = audioFile;
+            var behavior = htmlEvent.behavior;
+            print("Audio file: " + file);
+            createParticle(effectName, micSync, behavior, file);
+        // Handles audio file browsing event
+        } else if (htmlEvent.type === "chooseAudioFile") {
+            if (htmlEvent.value !== CLEAR_SELECTION_TEXT) {
+                audioFile = Window.browse("Choose an audio file");
+                if (audioFile !== null) {
+                    if (audioFile.indexOf(".wav") === -1) {
+                        Window.alert("Must be a .wav file type");
```

Comment:
Was this included in the test plan? I'm not sure Window.alert is the best way to display an error here.